File: 696. Count Binary Substrings.js
var countBinarySubstrings = function(s) {
    let start = 0;
    let end = s.length-1;
    let index = 0;
    let count = 0;
    let isCount = false;
    
    
    while(start < end){
        if(s.charAt(start) != s.charAt(start+1)){
            index = start;
            isCount = true;
            count++;
        } else if(isCount == true && index >= 0 && s.charAt(index) == s.charAt(index-1)){
            count++;
            index--;
        } else{
            isCount = false;
        }
        start++;
    }
    return count;
};

File: javascript/696. Count Binary Substrings.java
class Solution {
    public int countBinarySubstrings(String s) {
        int count = 0;
        int backwardIndex = 0;
        boolean isCount = false;
        
        for(int i = 0; i < s.length()-1; i++){
            if(s.charAt(i) != s.charAt(i+1)){
                isCount = true;
                backwardIndex = i;
                count++;
                continue;
            }
            if(isCount == true && backwardIndex > 0 && s.charAt(backwardIndex) == s.charAt(backwardIndex-1)){
                count++;
                backwardIndex --;
                continue;
            }
            isCount = false;
        }
        
        return count;
    }
}
